sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -713,9 +713,16 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
           {
             result.FileSize = std::stoll(node["contentLength"].get<std::string>());
           }
-          result.Owner = node["owner"].get<std::string>();
-          result.Group = node["group"].get<std::string>();
-          result.Permissions = node["permissions"].get<std::string>();
+          #define READ_STR(PROP, FIELD)                      \
+            if (node.contains(#PROP)) {                      \
+              result.FIELD = node[#PROP].get<std::string>(); \
+            }
+
+          READ_STR(owner, Owner)
```

Comment:
@Jinming-Hu  if Owner, Group and Permissions can be missing in the server-response, we should use Nullable for those right?  ( Otherwise how do I know server is not returning empty string v/s the default object init ).

Also, we have `json_option.hpp` in the internal layer from Core in case you want to use it. It has the `SetIfExists` method that you can call like
`SetIfExists(result.Owner, node, "owner")`   
The method would expect `result.Owner` to be optional.   We don't have one method for non-optionals becuase those values must be always present in the json.

Adding @ahsonkhan as FYI .  This can be useful to future auto-gen

Reply:
We also have an extra method to make it easier to parse from json string/numver to some non-string type.
For instance,  parsing a datetime from json number to a Nullable<Azure::DateTIme>:

```
JsonOptional::SetIfExists<int64_t, Azure::DateTime>(
      properties.UpdatedOn, fragment, UpdatedPropertyName, PosixTimeConverter::PosixTimeToDateTime);
```
The template takes the type we read from json `T` and the type we want to produce from it `V`.
Then the last argument is a callback which takes the `T` and returns the `V`  :)

Reply:
@vhvb1989 Those fields are always present in response.

This customer is not using the correct client, instead of using datalake clients, we should use blob clients for storage accounts without HNS enabled.

Reply:
@Jinming-Hu, as SDK provider, I think there are 3 options to make customer happier, :) (By the way, I'm not that sad, but definitely could be better)
1. option 1: fail datalake client completely if user doesn't enable HNS (I don't know what this is), and print error message with your above message.
2. option 2: make sure this is always true `Those fields are always present in response`, obviously it is not
3. option 3: make it optional instead of assert, and let customer decide if they want to continue when seeing these fields missing.

Frankly, the product we're testing is "Data Lake Gen2", and you suggest customer to not use datalake clients, it sounds wired to me.

This issue is not blocking me, but hope my comments help make it better in the future. Thanks for looking into the problem though, appreciate your time. And thanks @vhvb1989 for chiming in too.

Reply:
@shawncao The problem here is that you're not using the correct storage account.

If you want to use DataLake, you should create storage account with HNS enabled, and you can use either DataLake clients or blob clients in sdk to interact with storage service.